api/controllers/ChatController.js

```diff
@@ -0,0 +1,22 @@
+var chats = [];
+
+module.exports = {
+  subscribe(req, res) {
+    let params = req.allParams();
+    req.socket.join('chatting');
+    sails.io.sockets.in('chatting').emit('chats', chats);
+  },
+
+  new(req, res) {
+    let chat = req.allParams();
+
+    if (!chat.username) {
+      chat.username = 'Anonymous';
+    }
+
+    chats.push(chat);
+
+    res.send(204);
+    sails.io.sockets.in('chatting').emit('chats', chats);
```

Comment:
This can be updated to `sails.io.sockets.in(req.socket.id).emit('chats', chats);` to only send existing messages to the new user.